app/foodrle/tests/game.py

```diff
@@ -0,0 +1,163 @@
+    RED, YELLOW, GREEN, calorie_hint, taste_hint, Taste, Puzzle, is_guess_correct, collect_hints
+from foodrle.models import Country, Dishes, User, MainIngredient
+
+
+class TestCountryMethods(TestCase):
+    fixtures = ['foodrle.yaml']
+
+    def setUp(self):
+        self.random_countries = random.choices(list(Country.objects.all()), k=10)
+        self.opposite_bearings = [
+            ('north', 'south'),
+            ('east', 'west'),
+            ('northwest', 'southeast'),
+            ('northeast', 'southwest')
+        ]
+
+    def test_distance_equality(self):
+        """Test to ensure the distance from Country A to Country B
+            is the same as the distance from B to A
+        """
+        for country_a, country_b in itertools.product(self.random_countries, repeat=2):
+            if country_a != country_b:
+                self.assertEqual(distance(country_a, country_b), distance(country_b, country_a))
+
+    def test_bearing_parity(self):
+        """
+            Checks that there is parity in direction computation
+            (e.g., if a is north of b, b must be south of a)
+        """
+        for country_a, country_b in itertools.product(self.random_countries, repeat=2):
+            a_b_bearing = bearing(country_a, country_b)
+            b_a_bearing = bearing(country_b, country_a)
+            if country_a != country_b:
+                for bearing1, bearing2 in self.opposite_bearings:
+                    if a_b_bearing == bearing1:
+                        self.assertEqual(b_a_bearing, bearing2)
+                    if a_b_bearing == bearing2:
+                        self.assertEqual(b_a_bearing, bearing1)
+            else:
+                self.assertEqual(a_b_bearing, b_a_bearing)
+
+
+class TestPuzzleMethods(TestCase):
+    fixtures = ['foodrle.yaml']
+
+    def test_create_puzzle_answer(self):
+        user = User.objects.get(pk=1)
+        puzzle = create_puzzle_answer(user)
+        self.random_countries = random.choices(list(Country.objects.all()), k=10)
+
+        possible_dishes = [d.name for d in Dishes.objects.all()]
+        self.assertIn(puzzle.ans_dish.name, possible_dishes)
+
+    def test_submit_guess_game_over(self):
+        self.assertFalse(submit_guess(1, 'foo', 7))
+
+    def test_submit_guess_invalid_dish(self):
+        self.assertFalse(submit_guess(1, 'notafood', 1))
+
+    def test_submit_guess_correct(self):
+        self.assertTrue(submit_guess(1, 'pavlova', 1))
+
+
+class TestGenericMethods(TestCase):
+    fixtures = ['foodrle.yaml']
+
+    def test_find_guess_cnt_no_guesses(self):
+        guess_cnt = find_guess_cnt(1, 5)
+        self.assertEqual(guess_cnt, 1)
+
+    def test_find_guess_cnt_some_guesses(self):
+        user = User.objects.get(pk=1)
+        puzzle = create_puzzle_answer(user)
+        beginning_guess_cnt = 1
+        submit_guess(puzzle.id, 'slurm', beginning_guess_cnt)
```

Comment:
love the names, but this is failing because the guesses you are entering aren't valid dishes